sheeprl/utils/callback.py

```diff
@@ -0,0 +1,57 @@
+from typing import Any, Dict, Optional
+
+import torch
+from lightning.fabric import Fabric
+from lightning.fabric.plugins.collectives import TorchCollective
+
+from sheeprl.data.buffers import ReplayBuffer
+
+
+class CheckpointCallback:
+    def on_checkpoint_coupled(
```

Comment:
Maybe we can add here a docstring explaining how, when and where those callbacks should be called. 🐑